component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/service/LocalCacheService.java

```diff
@@ -17,71 +17,236 @@
 
 import java.io.ObjectStreamException;
 import java.io.Serializable;
+import java.util.List;
+import java.util.Map.Entry;
 import java.util.Objects;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.function.Predicate;
 import java.util.function.Supplier;
 
+import org.talend.sdk.component.api.configuration.Option;
 import org.talend.sdk.component.api.service.cache.LocalCache;
+import org.talend.sdk.component.api.service.configuration.Configuration;
 import org.talend.sdk.component.runtime.serialization.SerializableService;
 
-import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
 
-@AllArgsConstructor
-public class LocalCacheService implements LocalCache, Serializable {
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static java.util.stream.Collectors.toList;
 
+
+/**
+ * Implementation of LocalCache with in memory conccurent map.
+ * @param <T>   the type of cached data.
+ */
+@RequiredArgsConstructor
+public class LocalCacheService<T> implements LocalCache<T>, Serializable {
```

Comment:
misses a close callback which stops the executor if not null (so this leaks threads)

Think a @PreDestroy callback should work but ensure it with a test (checking threads before creating the local cache service, while it is active (+1 thread) and after the destruction (back to the same as before the creation)

Finally it will require to handle the state of the service (closed or not) to prevent the pool to be recreated and eviction to be submitted if the service is already closed.